onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -18,11 +19,22 @@
 #include "core/providers/qnn/builder/op_builder_factory.h"
 #include "core/providers/qnn/builder/qnn_def.h"
 #include "core/providers/qnn/builder/onnx_ctx_model_helper.h"
+#include "core/framework/run_options.h"
 
 namespace onnxruntime {
 
 constexpr const char* QNN = "QNN";
 
+static std::unique_ptr<std::vector<std::function<void()>>> s_run_on_unload_;
```

Comment:
Sorry I don't see how this s_run_on_unload_ is used? Once it is initialized in RunOnUnload(), how do we use these functions?

Reply:
Sorry, I missed another part of it. OnUnload. It's from provider_bridge_provider.cc. It's a because QNN EP was not implemented as shared lib.

---
In reply to: [1491942721](https://github.com/microsoft/onnxruntime/pull/19521#discussion_r1491942721) [](ancestors = 1491942721)